File: p2/p2.erl
#!/usr/bin/env escript

calc(N) ->
    calc(1, 2, 0, N).

calc(Prev, Curr, Sum, Max)
    when Sum >= Max ->
        Sum;

calc(Prev, Curr, Sum, Max)
    when Curr rem 2 == 0 ->
        calc(Curr, Curr + Prev, Sum + Curr, Max);

calc(Prev, Curr, Sum, Max) ->
    calc(Curr, Curr + Prev, Sum, Max).

main(_) ->
    Rslt = calc(4000000),
    io:format("~p", [Rslt]).

File: p2/p2.rs

fn main() {
    let mut prev = 1, curr = 2, rslt = 0;
    while curr <= 4000000 {
        if curr % 2 == 0 {
            rslt += curr
        }
        prev <-> curr;
        curr += prev;
    }

    io::println(fmt!("%?", rslt))
}
